src/fe-common/core/fe-common-core.c

```diff
@@ -462,25 +463,59 @@ void fe_common_core_finish_init(void)
 gboolean strarray_find_dest(char **array, const TEXT_DEST_REC *dest)
 {
 	g_return_val_if_fail(array != NULL, FALSE);
-
-	if (strarray_find(array, "*") != -1)
+	const char *type = module_find_id_str("WINDOW ITEM TYPE", dest->window->active->type);
+
+	if ((strarray_find(array, "*") != -1) || // we ignore all targets
+		(g_ascii_strcasecmp(type, "CHANNEL") == 0 && strarray_find(array, "#") != -1) || // we ignore all channels
```

Comment:
Having a small helper function taking a `TEXT_DEST_REC` and returning an enum depending on whether it is a channel, a dcc query or a regular one would make the code easier to read and avoid many string comparisons.